app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -877,20 +877,6 @@ class HomeFragment : Fragment() {
     }
 
     private fun navigateToSearch() {
-        // Dismisses the search dialog when the home content is scrolled
-        val recyclerView = sessionControlView!!.view
-        val listener = object : RecyclerView.OnScrollListener() {
-            override fun onScrollStateChanged(recyclerView: RecyclerView, newState: Int) {
-                super.onScrollStateChanged(recyclerView, newState)
-                if (newState == RecyclerView.SCROLL_STATE_DRAGGING || newState == RecyclerView.SCROLL_STATE_SETTLING) {
-                    findNavController().navigateUp()
-                    recyclerView.removeOnScrollListener(this)
-                }
-            }
-        }
-
-        recyclerView.addOnScrollListener(listener)
```

Comment:
I am just curious. How come we didn't need this onScroll handler anymore?

Reply:
Oh god I hope i didn't block the PR.

Reply:
It was a mistake to add the listener on the fragment side when the toolbar itself can handle it.  Also `OnScrollListener` was not the one we should be listening to.  This ignores the tap on the homescreen since it is not a scroll, and the first scroll is ignored because this is intercepting it.

Overall we shouldn't be using this.  My new change replaced this with just listening to the touch on the scrim area.

Reply:
Gotcha, appreciate the explanation and insight.